frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -398,19 +394,18 @@ export const createOrUpdateDeployImageResources = async (
     }
   } else if (!dryRun) {
     // Do not run serverless call during the dry run.
-    let imageStreamRepo: string = _.split(_.get(image, 'dockerImageReference', ''), '@')[0];
+    let imageStreamUrl: string = image?.dockerImageReference;
     if (registry === RegistryType.External) {
       const imageStreamResponse = await createOrUpdateImageStream(
         formData,
         dryRun,
         _.get(appResources, 'imageStream.data'),
         verb,
       );
-      imageStreamRepo = imageStreamResponse.status.dockerImageRepository;
+      const imageStreamRepo = imageStreamResponse.status.dockerImageRepository;
```

Comment:
@invincibleJai yesterday we discussed that in case of knative also for external registry we can get `imageRef` as `image?.dockerImageReference`. Did you find any use case in which this didnt work and we needed `dockerImageRepository` to get the correct image?

Reply:
yes @divyanshiGupta in  case of the private external registry we need that

Reply:
Okay, but as we saw yesterday we dont get the correct image in case of public external registry using `dockerImageRepository` and also in case of edit scenario. I will again test this against the use cases I have in mind. Will let you know if everything works fine.

Reply:
@divyanshiGupta pls, test against other scenarios and let me know